tools/toolchains/gcc.py

```diff
@@ -62,7 +62,7 @@ def __init__(self, target,  notify=None, macros=None, build_profile=None,
         elif target.core.startswith("Cortex-M33F"):
             self.cpu = ["-mcpu=cortex-m33"]
         elif target.core.startswith("Cortex-M33"):
-            self.cpu = ["-march=armv8-m.main"]
+            self.cpu = ["-mcpu=cortex-m33"]
```

Comment:
Cortex-M33 by default will add floating point and DSP instructions, Will have to add ‘+nofp’ and ‘+nodsp’ for Cortex-M33. Also we will need `+nodsp` for  `elif target.core.startswith("Cortex-M33F")` case

Reply:
Please change this to self.cpu = ["-mcpu=cortex-m33+nodsp+nofp"] - Default Cortex-M33 implementation without DSP and FP support